File: 404-左叶子之和.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    bool isLeaf(TreeNode* root)
    {
        return root->left==nullptr && root->right==nullptr;
    }

    int dfs(TreeNode* root)
    {
        int ans=0;
        if(root->left)
        {
            if(isLeaf(root->left))
                ans+=root->left->val;
            else
                ans+=dfs(root->left);
        }
        if(root->right && !isLeaf(root->right))
            ans+=dfs(root->right);
        return ans;
    }

    int sumOfLeftLeaves(TreeNode* root) {
        return root ? dfs(root) : 0;
    }
};

File: 404-左叶子之和.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def sumOfLeftLeaves(self, root: TreeNode) -> int:
        def isLeaf(root: TreeNode) -> bool:
            return root.left == None and root.right == None

        def dfs(root: TreeNode) -> int:
            ans = 0
            if root.left:
                if isLeaf(root.left):
                    ans += root.left.val
                else:
                    ans += dfs(root.left)
            if root.right and not isLeaf(root.right):
                ans = ans + dfs(root.right)
            return ans

        return dfs(root) if root else 0